pkg/kubelet/images/image_manager.go

```diff
@@ -73,19 +73,31 @@ func NewImageManager(recorder record.EventRecorder, imageService kubecontainer.I
 	}
 }
 
-// shouldPullImage returns whether we should pull an image according to
-// the presence and pull policy of the image.
-func shouldPullImage(pullPolicy v1.PullPolicy, imagePresent bool) bool {
-	if pullPolicy == v1.PullNever {
-		return false
+// imagePullPrecheck inspects the pull policy and checks for image presence accordingly,
+// returning (imageRef, err).
+func (m *imageManager) imagePullPrecheck(ctx context.Context, pullPolicy v1.PullPolicy, spec kubecontainer.ImageSpec) (string, error) {
+	// avoid fetching image if always pulling
+	if pullPolicy == v1.PullAlways {
+		return "", nil
 	}
 
-	if pullPolicy == v1.PullAlways ||
-		(pullPolicy == v1.PullIfNotPresent && (!imagePresent)) {
-		return true
+	// fetch imageRef to check image presence
+	imageRef, err := m.imageService.GetImageRef(ctx, spec)
+	if err != nil {
+		return "", ErrImageInspect
 	}
+	imagePresent := imageRef != ""
```

Comment:
no need for this var here. You can check it inside the "Never" if

Reply:
Done